### PR TITLE
Added initial autolabel option

### PR DESCRIPTION
**Pull request by ajturner**
First draft - let me know what do you think @abakirov 

Needs code clean up and to not break Pie charts. 

bar_chart___esri_cedar

bar_chart___esri_cedar

```json
  chart.show({
    elementId: "#chart-url",
    autolabels: true
  });
```

### CODE REVIEW

**Comment by abakirov** [comment]
@ajturner 
I somehow missed notification about this and noticed just now... This is cool, thanks!
Quick question: shouldn't `autolabels` option be always `true` or are there use cases where it's undesirable?

File: src/cedar.js [hunk 1]
```diff
@@ -383,6 +393,41 @@ Cedar.prototype._renderSpec = function(spec){
...
+
+    ['x','y'].forEach(function(axis, index) {
+      var angle = 0;
+      if (spec.axes[index].properties.labels.angle !== undefined) {
+        angle = spec.axes[index].properties.labels.angle.value;
+      }
+      if(spec.axes[index].type == 'y' ) {
+        angle = 100 - angle;
+      }      
+      spec.axes[index]["titleOffset"] = lengths[axis] * angle/100 * 8 + 30;
```
Comment (abakirov) [inline src/cedar.js]: magic numbers :wink:

**Comment by ajturner** [comment]
That's why it's 'initial' :blossom: and needs more development over time. But I wanted to get a working version for your prototype development.

`autolabel: true` - I'm back and forth on this. Probably should - at least for 'built in types' like bar, line, etc. when users design their own chart types this algorithm likely will not handle their custom layout.

But then that's not the normal (at this point). 

Regardless, I suggest you set it explicitly true for all your charts for now.

**Comment by abakirov** [comment]
I did a quick test on my end - this is definitely an improvement, thanks

**Comment by ajturner** [comment]
Anything else needed or can this be merged for now and we can improve in future PR?

**Comment by abakirov** [comment]
Please merge. In the future, I think we may want to come up with more accurate estimate algorithm for label widths.

**Pull request merged**